exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/__init__.py

```diff
@@ -211,7 +230,81 @@ def _translate_to_zipkin(self, spans: Sequence[Span]):
                 zipkin_span["parentId"] = format(span.parent.span_id, "016x")
 
             zipkin_spans.append(zipkin_span)
-        return zipkin_spans
+
+        return json.dumps(zipkin_spans)
```

Comment:
It would be better if this method returned a list of spans instead of final JSON and then another method encoded the list. This makes it very easy for other to extend an exporter and add features. Either that or we could extract per span translation logic into another function so people can extend that instead so the loop in this method looks like:

```python
zipkin_spans = [self._translate_to_zipkin_span(span) for span in spans]
```

Reply:
I'm not quite sure I see the use case - can you give an example?

Since we start off with our list of spans being passed into `ZipkinSpanExporter.export()` and convert it into a transport-specific format (v2.json, v2.proto), I would expect any desired modifications of the spans to occur prior to the `export()` call. The exporter translation logic adheres to the published zipkin API formats, so I don't see a scenario where those would need tweaked.

Reply:
Use case is where a vendor distribution of this package might have a specialized zipkin exporter that customizes some behavior or needs to fix a bug locally before it gets patched upstream. Smaller, self-container methods make it easier to sub-class and just change one part of the logic instead of copying an entire big function to just change a single line somewhere. 

Example: This line fixes a bug in Jaeger exporter in one of the distributions but because we crammed up whole logic in a single big function, the sub-class had to copy it all. https://github.com/signalfx/splunk-otel-python/blob/main/splunk_otel/exporters/jaeger.py#L105

Reply:
Thanks for the context. Considering the proposed options:

1. **Have the method return a list of spans** - not a very clean / performant approach since the custom wrapper code would have to re-traverse all of the spans to make the desired edits. I'm assuming this is why the jaeger patch didn't take this route even though it was possible since `_translate_to_jaeger()` does return a list of spans which could be modified in the `export()` method prior to sending?
2. **extract per span translation logic to separate func** - agree this is structurally cleaner, but practically speaking in both cases (jaeger / zipkin) the overrider is still going to be copy+pasting 95% of the overall logic, so the benefit seems low for how often this will occur (as you point out likely just temporary quick bug patching).

Reply:
Option 2 sounds good to me but no serious downsides. I was going to refactor most of these exporters to make this change anyway actually. Up to you.

Reply:
will leave for you since you're looking to do it across the board.